src/id_proofs.rs

```diff
@@ -0,0 +1,241 @@
+use concordium_base::{
+    common::VERSION_0,
+    id::{
+        constants::{ArCurve, AttributeKind, IpPairing},
+        id_proof_types::{ProofVersion, StatementWithContext},
+        types::IdentityObjectV1,
+    },
+};
+use key_derivation::CredentialContext;
+use serde::{Deserialize, Serialize};
+use uniffi::deps::anyhow::Context;
+use wallet_library::wallet::get_wallet;
+
+use crate::{
+    AttributeTag, Bytes, ConcordiumWalletCryptoError, ConvertError, GlobalContext, IdentityObject,
+    Network, Versioned,
+};
+
+/// For the case where the verifier wants the user to show the value of an
+/// attribute and prove that it is indeed the value inside the on-chain
+/// commitment. Since the verifier does not know the attribute value before
+/// seing the proof, the value is not present here.
+#[derive(Serialize, Deserialize, Clone)]
+pub struct RevealAttributeStatement<Tag: Clone> {
+    /// The attribute that the verifier wants the user to reveal.
+    #[serde(rename = "attributeTag")]
+    pub attribute_tag: Tag,
+}
+
+/// Serves as a uniFFI compatible bridge to [`concordium_base::id::id_proof_types::RevealAttributeStatement<ArCurve, AttributeTag>`]
+pub type RevealAttributeStatementV1 = RevealAttributeStatement<AttributeTag>;
+
+/// For the case where the verifier wants the user to prove that an attribute is
+/// in a range. The statement is that the attribute value lies in `[lower,
+/// upper)` in the scalar field.
+#[derive(Serialize, Deserialize, Clone)]
+pub struct AttributeInRangeStatement<Tag: Clone, Value: Clone> {
+    /// The attribute that the verifier wants the user to prove is in a range.
+    #[serde(rename = "attributeTag")]
+    pub attribute_tag: Tag,
+    /// The lower bound on the range.
+    #[serde(rename = "lower")]
+    pub lower: Value,
+    #[serde(rename = "upper")]
+    /// The upper bound of the range.
+    pub upper: Value,
+}
```

Comment:
Specifying `camelCase` can save some of the write-up. This applies to other large structs as well.

```suggestion
#[derive(Serialize, Deserialize, Clone)]
#[serde(rename_all = "camelCase")] 
pub struct AttributeInRangeStatement<Tag: Clone, Value: Clone> {
    /// The attribute that the verifier wants the user to prove is in a range.
    pub attribute_tag: Tag,
    /// The lower bound on the range.
    pub lower: Value,
    /// The upper bound of the range.
    pub upper: Value,
}
```